### ISSUE TITLE
Search through recipes given a keyword

### ISSUE THREAD

**Issue by JackKiefer** [JackKiefer/herms#36]
I'm thinking of something similar to `grep` but that searches instead through recipes. For now, just search through what `showRecipe` would typically display, we can get fancier later! 

The interaction might look something like this:

```
$ herms --search "seaweed"

4. Spicy Tempeh Sushi: * 4 sheets nori seaweed
4. Spicy Tempeh Sushi: (3) Place the rise on the seaweed sheet
7. Seaweed and Hummus Platter: | Seaweed and Hummus Platter | 
7. Seaweed and Hummus Platter: * Packaged seaweed sheets
7. Seaweed and Hummus Platter: (5) Put the seaweed on the platter
11. Goldfish Cracker Feast: (3) Place the sour straws in wavy patterns along the plate to emulate the look of seaweed
```

You can see here that the recipe index and recipe name are given followed by the relevant line of text that has been found. Sometimes it finds the word in the title, sometimes in the ingredients, sometimes in the directions. It just needs to be a simple "I found the string you're looking for on this line!"

Good luck! Throw questions this way if you have any :)

**Comment by renanpvaz** [comment]
Hey there, I think I could work on this. Any tips for a Haskell newbie?

**Comment by siddharthist** [comment]
@renanpvaz That sounds wonderful! Please comment here or email me if any questions come up. As you begin coding, you may want to take a look at #76, which makes some significant changes and should be merged soon(ish) (to be sure we don't end up with conflicting changes).

**Comment by dreamsmasher** [comment]
Early implementation in #123. However, colorization and bolding still need to be added, as well as support for multiple regular expressions at once.

Should we bold the titles of the matching recipes? Or the string matches?

**Comment by langston-barrett** [comment]
> as well as support for multiple regular expressions at once

What would be the advantage of multiple regexes (`rx1` and `rx2`) vs. a regex union like `(rx1|rx2)`?

> Should we bold the titles of the matching recipes? Or the string matches?

I'd say the string matches, but that's just personal preference. I'm not too picky as long as it looks good!

**Comment by dreamsmasher** [comment]
I think it's more user friendly to be able to specify multiple regexes in sequence, but if a user actually uses regex features instead of just strings, I figure they would know how to use unions as well. Just a thought.

I'll take a look into colorizing the string match output when I get a chance, really busy week :(

**Comment by langston-barrett** [comment]
> I think it's more user friendly to be able to specify multiple regexes in sequence, but if a user actually uses regex features instead of just strings, I figure they would know how to use unions as well. Just a thought.

Fair enough, that's basically what I expected but wanted to be sure I wasn't missing anything. I'm happy either way (if we have support for multiple regexes or not).

> I'll take a look into colorizing the string match output when I get a chance, really busy week :(

Sure thing, no need to hold up a nice new feature with window dressing.
